### PR TITLE
Zarr tests

### PR DESCRIPTION
**Pull request by kkoz**
First pass at ZarrPixelBufferTests

### CODE REVIEW

**Comment by chris-allan** [comment]
@kkoz: Green as of 0ba571f27ed6e57660f1e28c14b1d230cc5ebef6. You can relax the logging now as you desire.

File: src/main/java/com/glencoesoftware/omero/ms/image/region/ZarrPixelBuffer.java [hunk 1]
```diff
@@ -397,13 +422,18 @@ public PixelData getTile(Integer z, Integer c, Integer t, Integer x, Integer y,
...
         try {
+            checkBounds(x + w, y + h, z, c, t);
             int[] shape = new int[] { 1, 1, 1, h, w };
             int[] offsets = new int[] { t, c, z, y, x };
             byte[] asArray = getBytes(shape, offsets);
             PixelData d = new PixelData(
                     getPixelsType(), ByteBuffer.wrap(asArray));
             d.setOrder(ByteOrder.nativeOrder());
             return d;
+        } catch(DimensionsOutOfBoundsException e) {
```
Comment (chris-allan) [inline src/main/java/com/glencoesoftware/omero/ms/image/region/ZarrPixelBuffer.java:433]: Formatting

File: src/test/java/com/glencoesoftware/omero/ms/image/region/ZarrPixelBufferTest.java [hunk 2]
```diff
@@ -0,0 +1,395 @@
+package com.glencoesoftware.omero.ms.image.region;
```
Comment (chris-allan) [inline src/test/java/com/glencoesoftware/omero/ms/image/region/ZarrPixelBufferTest.java:1]: Needs a file header.

**Review by chris-allan — CHANGES_REQUESTED** [review]
Did you have problems with the float vs. double and autoboxing comparisons in the test cases?

**Comment by kkoz** [comment]
@chris-allan It wasn't really the autoboxing. Just that the JUnit assertEquals method for primitive floating point numbers requires a delta to be provided, so instead I converted to the objects types for comparison.

**Comment by chris-allan** [comment]
:+1:

In b31e6b0350ebd521d421232a6473555b5d28039f I've just simplified that a little so that there's no autoboxing confusion created by setting no delta. Effectively they're the same thing.

Let me know if you want to do anything else after the commits I've added and if not I'll get this merged.

**Comment by kkoz** [comment]
I think it's ready to be merged.

**Pull request merged**